insonmnia/miner/builder.go

```diff
@@ -139,8 +146,34 @@ func (b *MinerBuilder) Build() (miner *Miner, err error) {
 
 	log.G(ctx).Info("collected Hardware info", zap.Any("hardware", hardwareInfo))
 
-	grpcServer := grpc.NewServer(grpc.RPCCompressor(grpc.NewGZIPCompressor()), grpc.RPCDecompressor(grpc.NewGZIPDecompressor()))
+	var (
+		creds       credentials.TransportCredentials
+		certRotator util.HitlessCertRotator
+	)
+	if os.Getenv("GRPC_INSECURE") == "" {
+		var (
+			ethKey  *ecdsa.PrivateKey
+			TLSConf *tls.Config
+		)
+		if b.cfg.ETH() == nil || b.cfg.ETH().PrivateKey == "" {
+			cancel()
+			return nil, fmt.Errorf("either PrivateKey or GRPC_INSECURE environment variable must be set")
+		}
+		ethKey, err = ethcrypto.HexToECDSA(b.cfg.ETH().PrivateKey)
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+		// The rotator will be stopped by ctx
+		certRotator, TLSConf, err = util.NewHitlessCertRotator(ctx, ethKey)
+		if err != nil {
+			return nil, err
+		}
+		creds = util.NewTLS(TLSConf)
+	}
+	grpcServer := util.MakeGrpcServer(creds)
 
+	fmt.Println("SSSS", b.cfg.HubResources())
```

Comment:
"SSSS" is like snakes do? But this is not a Python :)

Reply:
Damn it) now you know, that I use println debugging.